### PR TITLE
NodeScopeResolver: call nodeCallback for all attribute-related nodes

### PR DESCRIPTION
**Pull request by janedbal**
Resolves https://github.com/phpstan/phpstan/issues/7071

### CODE REVIEW

**Comment by staabm** [comment]
should contain a regression test for the issue being fixed

**Comment by janedbal** [comment]
Tests added

**Comment by ondrejmirtes** [comment]
Please fix the build. Lint can be fixed by adding `// lint >= 7.4` comments, tests should be fixed by skipping on unsupported PHP versions. Thanks.

File: src/Analyser/NodeScopeResolver.php [hunk 1]
```diff
@@ -632,13 +621,7 @@ private function processStmtNode(
...
 			}
 
-			foreach ($stmt->attrGroups as $attrGroup) {
-				foreach ($attrGroup->attrs as $attr) {
-					foreach ($attr->args as $arg) {
-						$this->processExprNode($arg->value, $classScope, $nodeCallback, ExpressionContext::createDeep());
-					}
-				}
-			}
+			$this->processAttributeGroups($stmt->attrGroups, $scope, $nodeCallback);
```
Comment (ondrejmirtes) [inline src/Analyser/NodeScopeResolver.php]: You're calling this with the wrong scope. You need `$classScope` here.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by ondrejmirtes** [comment]
FYI this will conflict with your PR https://github.com/phpstan/phpstan-src/pull/1338, please update your code too.

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**